app/task_routes.py

```diff
@@ -0,0 +1,108 @@
+def get_all_tasks():
+        sort_query = request.args.get("sort")
+        title_query = request.args.get("title")
+        if sort_query == "asc":    
+            tasks = Task.query.order_by(Task.title).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title)).all()
+        elif sort_query:
+            return {
+                "details": f'Sort by "{sort_query}" is not an option'
+            }, 404
+        elif title_query:
+            tasks = Task.query.filter_by(title=title_query)
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = [task.task_to_json() for task in tasks]
+        return jsonify(tasks_response), 200
+
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return False
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def handle_single_task(task_id):
+    if not is_int(task_id):
+        return {
+        "details": "Task id must be an integer"
+        }, 404
+
+    task = Task.query.get(task_id)
+    if not task:
+        return jsonify(None), 404
+    
+    if request.method == "GET":
+        return {"task": task.task_to_json()}, 200
+    elif request.method == "PUT":
+        replace_task_data = request.get_json()
+        if replace_task_data.keys() >= {"title", "description", "completed_at"}:
+            task.title = replace_task_data["title"]
+            task.description = replace_task_data["description"]
+            task.completed_at = replace_task_data["completed_at"]
+            db.session.commit()
+            return {"task": task.task_to_json()}, 200
+        else: 
+            return {"details": "Invalid data"}, 400
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {
+            "details": f'Task {task.id} "{task.title}" successfully deleted'
+        }
+
+@tasks_bp.route("/<task_id>/<task_stutas>", methods=["PATCH"], strict_slashes=False)
+def update_task_status(task_id, task_stutas):
+    if not is_int(task_id):
+        return {
+        "details": "Task id must be an integer"
+        }, 404
+
+    task = Task.query.get(task_id)
+    if not task:
+        return jsonify(None), 404
+    
+    if task_stutas == "mark_complete":
+        task.completed_at = datetime.utcnow()
+        db.session.commit()
+        slack_api_url = "https://slack.com/api/chat.postMessage"
+        task_params = {
+            "channel": "task-notifications",
+            "text": f"Someone just completed the task {task.title}"
+        }
+        task_autho = {"Authorization": os.environ.get("SLACK_CHAT_POST_MESSAGE_TOKEN")}
+        requests.post(slack_api_url, json=task_params, headers=task_autho)
+        return  {"task": task.task_to_json()}, 200
+    elif task_stutas == "mark_incomplete":
+        task.completed_at = None
+        db.session.commit()
+        return  {"task": task.task_to_json()}, 200
+
```

Comment:
Making task_stutas a variable is an interesting solution!  There is a small bug here - if anything other than mark_complete or mark_incomplete is passed, this function will exit without a return, which will throw an error in Flask.  This can be handled by adding an `else` clause that returns a default or error response.